src/oxygen/base_handler.py

```diff
@@ -96,14 +98,27 @@ def _build_results(self, keyword, setup_keyword, teardown_keyword):
         setup_keyword: The special oxygen setup wrapper
         teardown_keyword: The special oxygen teardown wrapper
         '''
-        test = keyword.parent
-        test_results = self.parse_results(self.run_time_data)
-        end_time, result_suite = self._interface.result.build_suite(100000,
-                                                                   test_results)
+        parse_result_params = len(signature(self.parse_results).parameters)
+        if isinstance(self.run_time_data, tuple):
+            if parse_result_params > len(self.run_time_data):
```

Comment:
You mean like this?
```python
        if isinstance(self.run_time_data, tuple) and parse_result_params > len(self.run_time_data):
            raise MismatchArgumentException(f'parse_results expects {parse_result_params} arguments but got {len(self.run_time_data)}')
        elif parse_result_params > 1:
            raise MismatchArgumentException(f'parse_results expects {parse_result_params} arguments but got 1')
```
This code will fail the `multiple args` test (the 2nd one). Because when run_time_data is 2 and parse_result_params is also 2 then it will go into the elif part of the statement and throw second exception.

You would actually need to do this:
```python
        if isinstance(self.run_time_data, tuple) and parse_result_params > len(self.run_time_data):
            raise MismatchArgumentException(f'parse_results expects {parse_result_params} arguments but got {len(self.run_time_data)}')
        elif not isinstance(self.run_time_data, tuple) and parse_result_params > 1:
            raise MismatchArgumentException(f'parse_results expects {parse_result_params} arguments but got 1')

```

But now it is the choice of original solution which consists of 3 simple if-s or 2 complicated-ish if-s.